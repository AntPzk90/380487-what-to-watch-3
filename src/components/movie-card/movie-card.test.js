import React from 'react';
import renderer from 'react-test-renderer';
import MovieCard from './../movie-card/movie-card.jsx';
import {Provider} from "react-redux";
import configureStore from 'redux-mock-store';
import {Router} from "react-router-dom";
import history from "../../history.js";

const mockStore = configureStore([]);

const filmMock = {
  title: `Bohemian rhapsody`,
  src: `bohemian-rhapsody.jpg`,
  poster: `bohemian-rhapsody.jpg`,
  titlePoster: `bohemian-rhapsody.jpg`,
  genre: `Comedy`,
  releaseDate: `2015`,
  preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
};

it(`SnapshotTest MovieCard`, () => {
  const store = mockStore({
    film: filmMock,
  });

  const tree = renderer
  .create(
      <Provider store={store}>
        <Router
          history={history}
        >
        <MovieCard
          film = {filmMock}
        />
        </Router>
      </Provider>, {
        createNodeMock: () => {
          return {};
        }
      })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
