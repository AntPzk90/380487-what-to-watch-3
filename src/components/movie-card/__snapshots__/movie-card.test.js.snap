// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnapshotTest MovieCard 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <img
      alt="Fantastic Beasts: The Crimes of Grindelwald"
      height="175"
      src="img/bohemian-rhapsody.jpg"
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
    onClick={[Function]}
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
    >
      Bohemian rhapsody
    </a>
  </h3>
</article>
`;
